Makefile

```diff
@@ -16,18 +16,32 @@ VERSION := $(VERSION_MAJOR)$(if $(VERSION_MINOR),.$(VERSION_MINOR))$(if $(VERSIO
 
 COVERAGE_PATH ?= $(shell pwd)/.coverage
 
+GO_MAJOR_VER = $(shell $(GO) version | cut -c 14- | cut -d' ' -f1 | cut -d'.' -f1)
+GO_MINOR_VER = $(shell $(GO) version | cut -c 14- | cut -d' ' -f1 | cut -d'.' -f2)
+MIN_GO_MAJOR_VER = 1
+MIN_GO_MINOR_VER = 20
+GO_VALIDATION_ERR = Go version is not supported. Please update to at least $(MIN_GO_MAJOR_VER).$(MIN_GO_MINOR_VER)
+
+check-go-version:
+	@if [ $(GO_MAJOR_VER) -gt $(MIN_GO_MAJOR_VER) ]; then \
+		exit 0 ;\
+	elif [ $(GO_MAJOR_VER) -lt $(MIN_GO_MAJOR_VER) ]; then \
+		echo '$(GO_VALIDATION_ERR)';\
+		exit 1; \
+	elif [ $(GO_MINOR_VER) -lt $(MIN_GO_MINOR_VER) ] ; then \
+		echo '$(GO_VALIDATION_ERR)';\
+		exit 1; \
+	fi
+
 all: $(NAME)
```

Comment:
The `all` target should appear before `check-go-version`.

Reply:
ah, good point, thanks, done